packages/fx-core/src/component/debugHandler/appManifest.ts

```diff
@@ -141,8 +144,12 @@ export class AppManifestDebugHandler {
       if (projectSettingsResult.isErr()) {
         return err(projectSettingsResult.error);
       }
-      this.projectSettingsV3 = projectSettingsResult.value as ProjectSettingsV3;
 
+      // save project settings as the project id may be updated
+      const projectSettingsPath = getProjectSettingsPath(this.projectPath);
+      await fs.writeFile(projectSettingsPath, JSON.stringify(projectSettingsResult.value, null, 4));
```

Comment:
Is it possible to write project settings only if its ID changes? To make the impact as less as possible.

Reply:
I prefer always writing it in case there are any other magic changes in `loadProjectSettingsByProjectPath` ...